pkg/infrastructure/clusterapi/clusterapi.go

```diff
@@ -0,0 +1,363 @@
+	}
+
+	// Pass cluster kubeconfig and store it in; this is usually the role of a bootstrap provider.
+	{
+		key := client.ObjectKey{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+		}
+		cluster := &clusterv1.Cluster{}
+		if err := cl.Get(context.Background(), key, cluster); err != nil {
+			return fileList, err
+		}
+		// Create the secret.
+		clusterKubeconfig := clusterKubeconfigAsset.Files()[0].Data
+		secret := utilkubeconfig.GenerateSecret(cluster, clusterKubeconfig)
+		if err := cl.Create(context.Background(), secret); err != nil {
+			return fileList, err
+		}
+	}
+
+	// Wait for the load balancer to be ready by checking the control plane endpoint
+	// on the cluster object.
+	var cluster *clusterv1.Cluster
+	{
+		if err := wait.ExponentialBackoff(wait.Backoff{
+			Duration: time.Second * 10,
+			Factor:   float64(1.5),
+			Steps:    32,
+		}, func() (bool, error) {
+			c := &clusterv1.Cluster{}
+			if err := cl.Get(context.Background(), client.ObjectKey{
+				Name:      clusterID.InfraID,
+				Namespace: capiutils.Namespace,
+			}, c); err != nil {
+				if apierrors.IsNotFound(err) {
+					return false, nil
+				}
+				return false, err
+			}
+			cluster = c
+			return cluster.Status.InfrastructureReady, nil
+		}); err != nil {
+			return fileList, err
+		}
+		if cluster == nil {
+			return fileList, fmt.Errorf("error occurred during load balancer ready check")
+		}
+		if cluster.Spec.ControlPlaneEndpoint.Host == "" {
+			return fileList, fmt.Errorf("control plane endpoint is not set")
+		}
+	}
+
+	if p, ok := i.capiProvider.(InfraReadyProvider); ok {
+		infraReadyInput := InfraReadyInput{
+			Client:        cl,
+			InstallConfig: installConfig,
+		}
+
+		if err := p.InfraReady(ctx, infraReadyInput); err != nil {
+			return fileList, fmt.Errorf("failed provisioning resources after control plane available: %w", err)
+		}
+	}
+
+	bootstrapIgn, err := injectInstallInfo(bootstrapIgnAsset.Files()[0].Data)
+	if err != nil {
+		return nil, fmt.Errorf("unable to inject installation info: %w", err)
+	}
+	bootstrapIgnSecret := IgnitionSecret(clusterID.InfraID, "bootstrap", []byte(bootstrapIgn))
+	masterIgnSecret := IgnitionSecret(clusterID.InfraID, "bootstrap", masterIgnAsset.Files()[0].Data)
+
+	// The cloud-platform may need to override the default
+	// bootstrap ignition behavior.
+	if p, ok := i.capiProvider.(IgnitionProvider); ok {
+		ignInput := IgnitionInput{
+			BootstrapIgnData: []byte(bootstrapIgn),
+			InfraID:          clusterID.InfraID,
+			InstallConfig:    installConfig,
+		}
+
+		if bootstrapIgnSecret, err = p.Ignition(ctx, ignInput); err != nil {
+			return fileList, fmt.Errorf("failed creating bootstrap ignition secret: %w", err)
+		}
+	}
+	machineManifests = append(machineManifests, bootstrapIgnSecret, masterIgnSecret)
+
+	// Create the machine manifests.
+	for _, m := range machineManifests {
+		m.SetNamespace(capiutils.Namespace)
+		if err := cl.Create(context.Background(), m); err != nil {
```

Comment:
this line could use the context defined line 147
```suggestion
		if err := cl.Create(ctx, m); err != nil {
```

Reply:
Done. Thanks!